kube/registry_credentials_test.go

```diff
@@ -0,0 +1,123 @@
+package kube
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/SUSE/fissile/testhelpers"
+)
+
+func TestMakeRegistryCredentialsForKube(t *testing.T) {
+	assert := assert.New(t)
+
+	rcreds, err := MakeRegistryCredentials(ExportSettings{})
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	rcredsYAML, err := testhelpers.RenderNode(rcreds, nil)
+	// config - Helm only
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+apiVersion: "v1"
+data:
+  .dockercfg: ""
+kind: "Secret"
+metadata:
+  name: "registry-credentials"
+  labels:
+    skiff-role-name: "registry-credentials"
+type: "kubernetes.io/dockercfg"
+`
+	assert.Equal(expectedYAML, string(rcredsYAML))
+}
+
+func TestMakeRegistryCredentialsForHelmWithDefaults(t *testing.T) {
+	assert := assert.New(t)
+
+	rcreds, err := MakeRegistryCredentials(ExportSettings{
+		CreateHelmChart: true,
+	})
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	rcredsYAML, err := testhelpers.RenderNode(rcreds, nil)
+	// config - Helm only
+	// .Values.kube.registry.hostname
+	// .Values.kube.registry.username
+	// .Values.kube.registry.password
+	// json, base64 encoded
+
+	if !assert.NoError(err) {
+		return
+	}
+
+	// Notes: The base64 decodes to
+	// 	{%!q(<nil>):{"username":"","password":"","auth":"JSFzKDxuaWw+KTolIXMoPG5pbD4p"}}
```

Comment:
That looks suspiciously like a bug; `%!q(<nil>)` probably isn't valid JSON

Reply:
It is very much not valid json. It sort of demonstrates that the output of fissile lacks a `required` in the template or some such.